cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/store/DefaultMetadataStore.java

```diff
@@ -580,12 +582,30 @@ public SearchResults apply(MetadataDataset input) throws Exception {
   }
 
   @Override
-  public void rebuildIndexes(MetadataScope scope) throws DatasetManagementException, IOException {
+  public void rebuildIndexes(MetadataScope scope,
+                             RetryStrategy retryStrategy) {
     byte[] row = null;
-    while ((row = rebuildIndex(row, scope)) != null) {
+    while ((row = rebuildIndexesWithRetries(scope, row, retryStrategy)) != null) {
       LOG.debug("Completed a batch for rebuilding {} metadata indexes.", scope);
     }
+  }
 
+  private byte[] rebuildIndexesWithRetries(final MetadataScope scope,
```

Comment:
Ok, I am not sure how the old method was throwing IOException and DatasetManagementException, and why that is now not possible any more, simply because you now retry inside the loop. But not a biggie.